File: src/main/java/net/lomeli/lomlib/util/ResourceUtil.java
package net.lomeli.lomlib.util;

import java.io.File;

import net.minecraft.client.renderer.texture.TextureAtlasSprite;
import net.minecraft.client.renderer.texture.TextureMap;
import net.minecraft.util.ResourceLocation;

import net.minecraftforge.fml.client.FMLClientHandler;
import net.minecraftforge.fml.common.FMLCommonHandler;

public class ResourceUtil {

    public static ResourceLocation getResource(String modid, String resource) {
        return new ResourceLocation(modid, resource);
    }

    public static ResourceLocation getGuiResource(String modid, String gui) {
        if (!gui.endsWith(".png") && !gui.endsWith(".PNG"))
            gui += ".png";
        return getResource(modid, "textures/gui/" + gui);
    }

    public static ResourceLocation getEntityTexture(String modid, String texture) {
        if (!texture.endsWith(".png") && !texture.endsWith(".PNG"))
            texture += ".png";
        return getResource(modid, "textures/entities/" + texture);
    }

    public static ResourceLocation getModelTexture(String modid, String texture) {
        if (!texture.endsWith(".png") && !texture.endsWith(".PNG"))
            texture += ".png";
        return getResource(modid, "textures/model/" + texture);
    }

    public static TextureAtlasSprite getSprite(String modid, String resourceLoc) {
        return FMLClientHandler.instance().getClient().getTextureMapBlocks().getAtlasSprite(modid + ":" + resourceLoc);
    }

    public static File getModsFolder() {
        return FMLCommonHandler.instance().getMinecraftServerInstance().getFile("mods");
    }

    public static File getAssetsFolder() {
        return new File(FMLClientHandler.instance().getClient().mcDataDir, "assets");
    }

    public static File getConfigFolder() {
        return FMLCommonHandler.instance().getMinecraftServerInstance().getFile("config");
    }

    public static TextureAtlasSprite getSprite(ResourceLocation resource) {
        return resource != null ? getSprite(resource.getResourceDomain(), resource.getResourcePath()) : null;
    }

    public static void bindTexture(ResourceLocation texture) {
        FMLClientHandler.instance().getClient().renderEngine.bindTexture(texture);
    }

    public static boolean registerTexture(TextureMap map, ResourceLocation resource) {
        return map.setTextureEntry(resource.toString(), map.registerSprite(resource));
    }
}

File: src/main/kotlin/net/lomeli/lomlib/util/client/ResourceUtil.kt
package net.lomeli.lomlib.util.client

import java.io.File

import net.minecraft.client.renderer.texture.TextureAtlasSprite
import net.minecraft.client.renderer.texture.TextureMap
import net.minecraft.util.ResourceLocation

import net.minecraftforge.fml.client.FMLClientHandler
import net.minecraftforge.fml.common.FMLCommonHandler

object ResourceUtil {

    fun getResource(modid: String, resource: String): ResourceLocation = ResourceLocation(modid, resource)

    fun getGuiResource(modid: String, gui: String): ResourceLocation {
        return getResource(modid, "textures/gui/" + gui)
    }

    fun getEntityTexture(modid: String, texture: String): ResourceLocation {
        return getResource(modid, "textures/entities/" + texture)
    }

    fun getModelTexture(modid: String, texture: String): ResourceLocation {
        return getResource(modid, "textures/model/" + texture)
    }

    fun getSprite(modid: String, resourceLoc: String): TextureAtlasSprite = FMLClientHandler.instance().client.textureMapBlocks.getAtlasSprite(modid + ":" + resourceLoc)

    val modsFolder: File
        get() = FMLCommonHandler.instance().minecraftServerInstance.getFile("mods")

    val assetsFolder: File
        get() = File(FMLClientHandler.instance().client.mcDataDir, "assets")

    val configFolder: File
        get() = FMLCommonHandler.instance().minecraftServerInstance.getFile("config")

    fun getSprite(resource: ResourceLocation?): TextureAtlasSprite? = if (resource != null) getSprite(resource.resourceDomain, resource.resourcePath) else null

    fun bindTexture(texture: ResourceLocation) {
        FMLClientHandler.instance().client.renderEngine.bindTexture(texture)
    }

    fun registerTexture(map: TextureMap, resource: ResourceLocation): Boolean = map.setTextureEntry(map.registerSprite(resource))
}